component/fetcher.go

```diff
@@ -51,12 +51,15 @@ func makeFetcher(
 		recordCounterMetric.Add(float64(len(batch)))
 		log.WithField("batch_size", len(batch)).
 			Infof("fetched records")
-
+		if pulse.Number < s.currentHeavyPN {
+			return &raw{pulse: pulse, batch: batch, shouldIterateFrom: shouldIterateFrom}
```

Comment:
Why we don't set `currentHeavyPN` here?

Reply:
yep, you need to return raw.currentHeavyPN to fastForwardInterwal works properly

Reply:
Done